add inception-v3 model

ebenolson opened the pull request:
Waiting on https://github.com/Lasagne/Lasagne/pull/467 to merge.

Any ideas on cleaner ways to parametrize the inception modules welcome.

f0k commented:
Cool! Looks good to me -- but is there any way to link the `inceptionX` functions to the figures in the paper (I've just skimmed it), to make it easier to follow?

f0k commented on modelzoo/inception_v3.py:
```diff
@@ -0,0 +1,200 @@
+
+
+def avg_pool(input_layer, **kwargs):
+    # hack to work around https://github.com/Theano/Theano/issues/3776
```
Has been shown to be a cuDNN v3 bug -- maybe use `average_exc_pad` now and live with the fact that it requires cuDNN v4? Theano gives an error if that mode is used with v3, so people will know.

f0k commented:
Regarding the changes in batch normalization, I've sketched how to convert the model: https://github.com/Lasagne/Lasagne/pull/467#issuecomment-170989499. You'd probably need to unpickle it once, convert, pickle again and reupload to S3. Let me know if you need help!

ebenolson commented:
Ok, I've updated the pickle, verified that it works with cuDNN v4 and added a few comments. Any other suggestions? If not I'll squash and merge tomorrow.

f0k commented:
LGTM :+1:

jmetzen commented:
Did anyone test the pretrained weights for Inceptionv3 on ImageNet validation? I've adapted the notebook "ImageNet Pretrained Network (VGG_S).ipynb" and haven't obtained good results ("honeycomb" and "website" are predicted as top classes typically).

ebenolson commented:
Works for me: https://gist.github.com/ebenolson/57979cdf6a42be21f2c9

jmetzen commented:
Thanks for the example! That was essentially what I was trying. I've noticed that I get the same (correct) result when running the example on the CPU but completely different results when running on GPU (also with your notebook). Did you run on GPU? Any ideas what might cause this issue?

jmetzen commented:
Never mind, I tried it on a newly installed CUDA/cudnn environment and got the correct results also on GPU. Not sure what caused the issue, though...
